src/pharmpy/plugins/nonmem/run.py

```diff
@@ -40,18 +41,46 @@ def execute_model(model):
             fp.write(f"#!/bin/sh\ncd {path}\n{' '.join(args)}\n")
         os.chmod(path / 'cdwrapper', 0o744)
         cmd = str(path / 'cdwrapper')
-    subprocess.call(
-        [cmd], stdin=subprocess.DEVNULL, stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL
-    )
+
+    stdout = path / 'stdout'
+    stderr = path / 'stderr'
+
+    with open(stdout, "wb") as out, open(stderr, "wb") as err:
+        result = subprocess.run([cmd], stdin=subprocess.DEVNULL, stderr=err, stdout=out)
+
+    metadata = {
+        'plugin': 'nonmem',
+        'path': str(path),
+    }
+
+    plugin = {
+        'commands': [
+            {
+                'args': args,
+                'returncode': result.returncode,
+                'wrapper': cmd,
+            }
+        ]
+    }
+
     database.store_model(model)
     database.store_local_file(model, (path / basepath).with_suffix('.lst'))
     database.store_local_file(model, (path / basepath).with_suffix('.ext'))
     database.store_local_file(model, (path / basepath).with_suffix('.phi'))
     database.store_local_file(model, (path / basepath).with_suffix('.cov'))
     database.store_local_file(model, (path / basepath).with_suffix('.cor'))
     database.store_local_file(model, (path / basepath).with_suffix('.coi'))
+
     for rec in model.control_stream.get_records('TABLE'):
         database.store_local_file(model, path / rec.path)
+
+    database.store_optional_local_file(model, stdout)
+    database.store_optional_local_file(model, stderr)
+    database.store_optional_json(model, plugin, 'nonmem.json')
```

Comment:
Do we need two json-files and two methods? Could `store_metadata` suffice?

Reply:
This is to separate plugin-generic stuff from plugin-specific stuff. One thing that can be done to reduce the number of methods is to first write the JSON to a local file, then use the existing `store_optional_local_file`.